opm/autodiff/WellInterface_impl.hpp

```diff
@@ -641,30 +641,72 @@ namespace Opm
     WellInterface<TypeTag>::
     updateWellTestState(const WellState& well_state,
                         const double& simulationTime,
-                        WellTestState& wellTestState,
-                        const bool& writeMessageToOPMLog) const
+                        const bool& writeMessageToOPMLog,
+                        WellTestState& wellTestState) const
     {
-        // economic limits only apply for production wells.
+        // currently, we only updateWellTestState for producers
         if (wellType() != PRODUCER) {
             return;
         }
 
-        // flag to check if the mim oil/gas rate limit is violated
-        bool rate_limit_violated = false;
+        // updating well test state based on physical (THP/BHP) limits.
+        updateWellTestStatePhysical(well_state, simulationTime, writeMessageToOPMLog, wellTestState);
+
+        // updating well test state based on Economic limits.
+        updateWellTestStateEconomic(well_state, simulationTime, writeMessageToOPMLog, wellTestState);
+    }
+
+
+
+
+
+    template<typename TypeTag>
+    void
+    WellInterface<TypeTag>::
+    updateWellTestStatePhysical(const WellState& /* well_state */,
+                                const double /* simulation_time */,
+                                const bool /* write_message_to_opmlog */,
+                                WellTestState& /* well_test_state */) const
+    {
+        // TODO: other facilities are required to make the following action work
```

Comment:
This PR is created based on the WIP branch. I did not realize other facilities are required for the `updateWellTestStatePhysical`, which makes this function an empty function.  Please let me know if you think I should remove this function from this PR.